### PR TITLE
Docker remove georchestra base

### PR DESCRIPTION
**Pull request by pmauduit**
The goal of this PR is to remove an extra layer with the `georchestra/base`. It also adds a definition of the volumes in order to allow data persistence.

The use of a georchestra/base image actually introduces some complexity to customize images afterwards, and forces the image to be generated with the complete datadir copied into /etc/georchestra which is unecessary.

The PR addresses also the need for people willing to use docker and already have a git repository to use it instead of the default one, by issuing the following command:

```
mvn clean package docker:build -DdockerDatadirScmUrl=scm:git:git@example.com:myprivaterepo/datadir.git -DdockerDatadirScmVersion=mycustombranch
```

Note: when merging this into master in the next cascade merge, it will require to modify the poms to change the `dockerDatadirScmVersion` to `docker-official` instead of `docker-15.12`.

Note2: once the PR onto geonetwork related to this one (https://github.com/georchestra/geonetwork/pull/36), it will require to bump the geonetwork submodule version.

Note3: the previous behaviour (mvn clean package docker:build ...) should stay iso-functional if this PR is merged, the main added value is that the parameters `-Dvariable=...` have been added for convenience.

Tests status (runtime on locally generated containers):
- [x] cas
- [x] header
- [x] mapfishapp
- [x] extractorapp
- [x] analytics
- [x] security-proxy
- [x] geowebcache
- [x] ldapadmin
- [x] geoserver
- [x] geonetwork 2
- [x] geonetwork 3.0.x
- [x] catalogapp
- [x] downloadform

Test protocol:

- `mvn clean package docker:build -Pdocker` on each module
- `docker run georchestra/<module>`
- check the docker output
- connect onto the container and check that the `/etc/georchestra` exists as expected, and contains only the subdirectory for the given module.

### CODE REVIEW

File: geoserver/webapp/src/docker/Dockerfile [hunk 1]
```diff
@@ -1,8 +1,17 @@
...
 
 ADD . /
 ADD https://github.com/georchestra/geoserver_minimal_datadir/archive/master.zip /tmp/datadir.zip
 
-RUN chmod +x /root/start.sh
+VOLUME [ "/var/local/geoserver", "/var/local/geodata" ]
+
+ENTRYPOINT [ "/docker-entrypoint.sh" ]
+CMD [ "java","-Djava.io.tmpdir=/tmp/jetty", "-Dgeorchestra.datadir=/etc/georchestra",                    \
+      "-DGEOSERVER_DATA_DIR=/var/local/geoserver/", "-Dgeofence.dir=/etc/georchestra/geoserver/geofence" \
```
Comment (Vampouille) [inline geoserver/webapp/src/docker/Dockerfile]: missing comma

**Comment by Vampouille** [comment]
LGTM with a comma at the end of : geoserver/webapp/src/docker/Dockerfile L15

File: geoserver/webapp/src/docker/Dockerfile (hunk 1 above)
Comment (pmauduit) [inline geoserver/webapp/src/docker/Dockerfile]: Thanks !

**Comment by fvanderbiest** [comment]
Please update https://github.com/georchestra/georchestra/blob/15.12/RELEASE_PROCEDURE.md also in this PR, since the poms now reference the current branch via `<dockerDatadirScmVersion>docker-15.12</dockerDatadirScmVersion>`

File: geoserver/webapp/src/docker/Dockerfile [hunk 2]
```diff
@@ -1,8 +1,17 @@
...
 
 ADD . /
 ADD https://github.com/georchestra/geoserver_minimal_datadir/archive/master.zip /tmp/datadir.zip
 
-RUN chmod +x /root/start.sh
+VOLUME [ "/var/local/geoserver", "/var/local/geodata" ]
+
+ENTRYPOINT [ "/docker-entrypoint.sh" ]
+CMD [ "java","-Djava.io.tmpdir=/tmp/jetty", "-Dgeorchestra.datadir=/etc/georchestra",                     \
+      "-DGEOSERVER_DATA_DIR=/var/local/geoserver/", "-Dgeofence.dir=/etc/georchestra/geoserver/geofence", \
```
Comment (fvanderbiest) [inline geoserver/webapp/src/docker/Dockerfile]: missing volume and option here : GEOWEBCACHE_CACHE_DIR (`/var/local/geowebcache`)

File: geoserver/webapp/src/docker/Dockerfile [hunk 3]
```diff
@@ -1,8 +1,17 @@
-FROM georchestra/base
+FROM jetty:9.3-jre8
+
+
+RUN apt-get update && \
+    apt-get install -y unzip && \
+    rm -rf /var/lib/apt/lists/*
 
 ADD . /
 ADD https://github.com/georchestra/geoserver_minimal_datadir/archive/master.zip /tmp/datadir.zip
```
Comment (fvanderbiest) [inline geoserver/webapp/src/docker/Dockerfile]: We should probably create a 15.12 branch in this repo, and target this branch here.

File: geowebcache-webapp/src/docker/Dockerfile [hunk 4]
```diff
@@ -1,3 +1,8 @@
-FROM georchestra/base
+FROM jetty:9.3-jre8
+
+ADD . /
+
+RUN java -jar "$JETTY_HOME/start.jar" --add-to-startd=jmx,jmx-remote,stats
+
+CMD ["java","-Djava.io.tmpdir=/tmp/jetty", "-Dgeorchestra.datadir=/etc/georchestra","-jar","/usr/local/jetty/start.jar"]
```
Comment (fvanderbiest) [inline geowebcache-webapp/src/docker/Dockerfile]: Missing volume for the cache (as above)

**Comment by fvanderbiest** [comment]
LGTM !

File: geoserver/webapp/src/docker/Dockerfile (hunk 3 above)
Comment (pmauduit) [inline geoserver/webapp/src/docker/Dockerfile]: Branch created, fix commited

File: geoserver/webapp/src/docker/Dockerfile (hunk 2 above)
Comment (pmauduit) [inline geoserver/webapp/src/docker/Dockerfile]: Done

File: geowebcache-webapp/src/docker/Dockerfile (hunk 4 above)
Comment (pmauduit) [inline geowebcache-webapp/src/docker/Dockerfile]: done

**Comment by fvanderbiest** [comment]
Please merge

**Pull request merged**